classlib/src/main/java/org/teavm/classlib/java/lang/TInteger.java

```diff
@@ -316,11 +316,11 @@ public static int numberOfTrailingZeros(int i) {
     }
 
     public static int highestOneBit(int i) {
-        return 0x80000000 >>> numberOfLeadingZeros(i);
+        return i & (0x80000000 >>> numberOfLeadingZeros(i));
```

Comment:
Not sure why this `i &` necessary here. The left operand for shift is still one bit.

Reply:
This is needed for case i == 0. Added test which covers this case.